cfme/tests/cli/test_appliance_console.py

```diff
@@ -744,27 +745,42 @@ def test_appliance_console_restart():
         testSteps:
             1. 'ap' launches appliance_console.
             2. RETURN clears info screen.
-            3. '16' Stop EVM Server Processes.
-            4. 'Y' Stop CFME Server Gracefully.
-            5. Wait for some time.
-            6. '17' Start EVM Server Processes.
-            7. 'Y' to Start CFME server.
-            8. Wait for some time.
-            9. '21' Quit from appliance_console menu.
-            10. Open appliance IP into Browser.
+            3. '18' restart appliance.
+            4. '1' Restart.
+            5. Verify UI (Open appliance IP into Browser.)
         expectedResults:
             1.
             2.
             3.
             4.
-            5. Cross-check service stopped.
-            6.
-            7.
-            8. Cross-check service started.
-            9.
-            10. Confirm that Appliance is running into browser.
+            5. Confirm that Appliance is running into browser.
     """
-    pass
+    appliance = temp_appliance_preconfig_funcscope
+    old_uptime = appliance.ssh_client.uptime()
+    restart_appliance_cmd = ("ap", RETURN, "18", "1", "Y")
```

Comment:
I would recommend to start using `SSHExpect` to validate the messages from appliance_conosole. Whithout the validation you do not know what command you are actually ordering as the menu changes and also you may have hard time figuring out which command was the incorrect one.

Reply:
`run_commands` function returns one list so we can check in which order commands executed and what output we got from commands.

Reply:
@dgaikwad the issue that @jaryn is raising is a bit different, its about a level of assertion that `18`, `1`, `Y` inputs are being sent against an expected prompt.  SSHExpect handles failing early if an expected prompt is not displayed - so it doesn't just continue sending inputs against an unknown state.

See here for an example of its use:
https://github.com/ManageIQ/integration_tests/blob/1943da5de8ced139ad25bbbb20852aee5831a379/cfme/utils/appliance/console.py#L114

Reply:
Tried to use SSHExpect but TC was failing:
Below is the code which I tried:


```

# Restarting the appliance
with SSHExpect(appliance) as interaction:
    interaction.send('ap')
    interaction.answer('Press any key to continue.', '', timeout=30)
    interaction.answer('Choose the advanced setting: ', app_con_menu["restart_app"], timeout=30)
    interaction.answer('Choose the restart option: ', '1', timeout=30)

    interaction.answer('Are you sure you want to restart the appliance now? (Y/N):, 'Y', timeout=30)
```

I was getting below error:
```
>               + textwrap.indent(current_output, '> '))
E           cfme.exceptions.SSHExpectTimeoutError: Timeouted when waiting for 'Are you sure you want to restart the appliance now? (Y/N):'. Current output:
E           > 1
E           > Are you sure you want to restart the appliance now? (Y/N):

cfme/utils/ssh_expect.py:40: SSHExpectTimeoutError

```

After discussion with @jaryn , we decided to keep the previous code due to SSHExpect issue. . Somehow SSHExpect was not able to match expected string.

Reply:
There is some misunderstanding. I think when we were discussing, we reached conclusion that you managed to make the SSHExpect match the string after using `re.escape`.

The problem was in the waiting for shutdown (quoting otputs from our the discussion):

```
            interaction.answer('Choose the restart option: ', '1', timeout=30)
            interaction.answer(re.escape('Are you sure you want to restart the appliance now? (Y/N): '), 'Y', timeout=30)
    
    
    
        wait_for(
            net_check, num_sec=180, func_args=["22", appliance.hostname, True, 3],
            message='waiting to shutdown machine',
>           fail_condition=True
        )
E       wait_for.TimedOutError: Could not do 'waiting to shutdown machine' at ... cfme/utils/net.py:61 in time
```
I don't want to make a pressure on Devidas to use the SSHExpect. I would like him to consider it.

Reply:
@jaryn you are right, I have not got any error while executing the commands but the machine was not going down,
```
        # Restarting the appliance
        with SSHExpect(appliance) as interaction:
            interaction.send('ap')
            interaction.answer('Press any key to continue.', '', timeout=30)
            interaction.answer('Choose the advanced setting: ', app_con_menu["restart_app"], timeout=30)
            interaction.answer('Choose the restart option: ', '1', timeout=30)
    
            interaction.answer(re.escape('Are you sure you want to restart the appliance now? (Y/N): '), 'Y', timeout=30)
    
    
        wait_for(
            net_check, num_sec=180, func_args=["22", appliance.hostname, True, 3],
            message='waiting to shutdown machine',
>           fail_condition=True
        )
E       wait_for.TimedOutError: Could not do 'waiting to shutdown machine' at /home/dgaikwad/redhat/integration_tests/cfme/utils/net.py:61 in time

cfme/tests/cli/test_appliance_console.py:805: TimedOutError

```